streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/IQv2EndpointToPartitionsIntegrationTest.java

```diff
@@ -0,0 +1,229 @@
+                        return threadMetadata.activeTasks().size() == 2 && threadMetadata.standbyTasks().size() == expectedStandbyTopicPartitionCount;
+                    }, TestUtils.DEFAULT_MAX_WAIT_MS,
+                            "KafkaStreams one never released active tasks and received standby task");
+
+                    waitForCondition(() -> {
+                        final ThreadMetadata threadMetadata = streamsTwo.metadataForLocalThreads().iterator().next();
+                        return threadMetadata.activeTasks().size() == 2 && threadMetadata.standbyTasks().size() == expectedStandbyTopicPartitionCount;
+                    }, TestUtils.DEFAULT_MAX_WAIT_MS,
+                            "KafkaStreams two never received active tasks and standby");
+
+                    waitForCondition(() -> {
+                        final List<StreamsMetadata> metadata = new ArrayList<>(streamsTwo.metadataForAllStreamsClients());
+                        return metadata.size() == 2 &&
+                               metadata.get(0).standbyTopicPartitions().size() == expectedStandbyTopicPartitionCount &&
+                               metadata.get(1).standbyTopicPartitions().size() == expectedStandbyTopicPartitionCount;
+                    }, TestUtils.DEFAULT_MAX_WAIT_MS,
+                            "Kafka Streams clients 1 and 2 never got metadata about standby tasks");
+
+                    final List<StreamsMetadata> allClientMetadataUpdated = new ArrayList<>(streamsTwo.metadataForAllStreamsClients());
+                    final StreamsMetadata streamsOneMetadataOneUpdated = allClientMetadataUpdated.get(0);
+                    final Set<TopicPartition> streamsOneTopicPartitions = streamsOneMetadataOneUpdated.topicPartitions();
+                    assertEquals(2020, streamsOneMetadataOneUpdated.hostInfo().port());
+                    assertEquals(2, streamsOneTopicPartitions.size());
+                    final long streamsOneRepartitionTopicTaskCount = streamsOneTopicPartitions.stream().filter(tp -> tp.topic().contains("-repartition")).count();
+                    final long streamsOneSourceTopicTaskCount = streamsOneTopicPartitions.stream().filter(tp -> tp.topic().contains("-input-two")).count();
+                    assertEquals(1, streamsOneRepartitionTopicTaskCount);
+                    assertEquals(1, streamsOneSourceTopicTaskCount);
+
+                    final StreamsMetadata streamsOneMetadataTwo = allClientMetadataUpdated.get(1);
+                    final Set<TopicPartition> streamsTwoTopicPartitions = streamsOneMetadataTwo.topicPartitions();
+                    assertEquals(3030, streamsOneMetadataTwo.hostInfo().port());
+                    assertEquals(2, streamsTwoTopicPartitions.size());
+                    assertEquals(expectedStandbyTopicPartitionCount, streamsOneMetadataTwo.standbyTopicPartitions().size());
```

Comment:
Should we check the contents of the standbyTopicPartitions? The javadoc states that they are changelog partitions, but it looks like we are not using changelog topics (up in the implementation). I am utterly confused now which topic partitions are included in which set now.

Can we also check the result of `standbyStateStoreNames`? It seems it is derived from `standbyTopicPartitions`, it should run the same logic in both protocols.

Reply:
Yes, I can update the test to verify `standbyStateStoreNames`.  

> The javadoc states that they are changelog partitions

I believe the javadoc is incorrect. I think confusion lies in the fact that standby tasks will **_consume_** from changelog topic partitions (via the restore consumer), but for IQ from standbys you'll need to query the input topic partitions (repartition ones if there's been a key changing operation for the stateful operation).  What's critical is that subtopology partitions and partitions of the changelog topics match, although I'm not sure if there's a practical way to compare those in a test - but I'll look into.

I'll do a PR to update the javadoc.

Reply:
> Can we also check the result of standbyStateStoreNames? It seems it is derived from standbyTopicPartitions

I glossed over this before, but I don't believe that's the case - where do you see that?

Reply:
> > It seems it is derived from standbyTopicPartitions
> 
> I glossed over this before, but I don't believe that's the case - where do you see that?

https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java#L392 I was looking at this - it seems like it's derived from `standbyPartitionHostMap`